src/layout.rs

```diff
@@ -123,16 +123,28 @@ impl Pinnacle {
         let tiled_windows = windows_on_foc_tags
             .iter()
             .filter(|win| !win.is_x11_override_redirect())
-            .filter(|win| win.with_state(|state| state.layout_mode.is_tiled()))
+            .filter(|win| {
+                win.with_state(|state| {
+                    state.layout_mode.is_tiled() || state.layout_mode.is_spilled()
+                })
+            })
             .cloned();
 
         let Some(output_geo) = self.space.output_geometry(output) else {
             warn!("Cannot update_windows_from_tree without output geo");
-            return;
+            return Vec::new();
         };
 
         let non_exclusive_geo = layer_map_for_output(output).non_exclusive_zone();
 
+        // TODO: I'm not a huge fan of doing it that way, but until
+        // State.update_window_layout_mode_and_layout gets refactored, it's best to defer setting
+        // the layout since we also need to compute the window geometry. This is a POC and I'm lazy
```

Comment:
This feels like a hack, and it very much is, but the only way to call the function to properly handling switching to a pseudo floating mode without issue was to return the vector of spilled windows to the calling scope. 

That being said, I think we could get away with just setting the Spilled layout mode here. Since it acts like a floating behavior, we don't need re-compute the layout anyway, so we might not need the full transaction handling logic from that scope.

If that's true, maybe we can just extract the logic to compute the new geometry form State.update_window_layout_mode_and_layout.

Any thought on this ?

Reply:
Done with [a96d222](https://github.com/pinnacle-comp/pinnacle/pull/326/commits/a96d2223f220ae66c3eb5f428b1663f901ef9075)